### PR TITLE
Add module name parameter for workflows and tasks

### PR DESCRIPTION
**Pull request by YmirKhang**
# TL;DR
This PR adds an optional parameter to workflow and task types to make the module name explicitly available.

```python
@task(
    cache_version="1",
    container_image="{{.images.default.fqn}}:{{.images.default.version}}",
    module_name="<app.workflows.workflow_file>",
)
def my_task() -> None:
...

...
@workflow(module_name="<app.workflows.workflow_file>")
def my_workflow() -> None:
```

So that it can be avoided that the workflows and tasks default to `__main__.my_task` and `__main__.my_workflow`, by overriding module name in tasks and workflows

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [X] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Added extra optional parameter `module_name` to workflow and task definitions.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1813

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by YmirKhang** [comment]
This is just an implementation to make it work with a self contained file. This was not tested, also other side effects are not considered. But since the parameter is optional it should be backward compatible.
